### PR TITLE
Create & edit organization and contacts

### PR DESCRIPTION
**Pull request by defectiveAi**

### CODE REVIEW

**Review by dzhavat — CHANGES_REQUESTED** [review]
praise: Good work so far! I bet RxJS is your favorite by now 😛

File: libs/dh/market-participant/feature-organization/src/lib/dh-market-participant-organization.component.html [hunk 1]
```diff
@@ -15,123 +15,132 @@
...
+        </mat-cell>
+      </ng-container>
+
+      <!-- Actor grid areas -->
+      <ng-container matColumnDef="{{ this.columnIds[4] }}">
+        <mat-header-cell *matHeaderCellDef>
+          {{ t("columns.gridAreas") }}
+        </mat-header-cell>
+        <mat-cell *matCellDef="let element">
+          {{ element.actor?.gridAreas ?? "-" }}
```
[the remark this answers is not in the document]
Comment (defectiveAi) [reply libs/dh/market-participant/feature-organization/src/lib/dh-market-participant-organization.component.html:98]: gridArea is indeed missing at the moment, but will be added in the future. We still do want to show an empty column though, as it is currently the correct state. We have added a comment.

File: libs/dh/market-participant/data-access-api/src/lib/dh-market-participant-data-access-api.store.ts [hunk 2]
```diff
@@ -15,55 +15,79 @@
...
-  organization: Partial<OrganizationDto>;
-  actor?: Partial<ActorDto>;
+export interface OverviewRow {
+  organization: OrganizationDto;
+  actor?: ActorDto;
 }
 
 interface MarketParticipantState {
   isLoading: boolean;
-  organizations: OrganizationWithActor[];
```
[the remark this answers is not in the document]
Comment (defectiveAi) [reply libs/dh/market-participant/data-access-api/src/lib/dh-market-participant-data-access-api.store.ts:33]: We refactored, it is saved once again.

File: libs/dh/market-participant/feature-edit-organization/src/lib/dh-market-participant-edit-organization/master-data/dh-market-participant-organization-master-data.component.html [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+-->
+<ng-container
+  *transloco="let t; read: 'marketParticipant.organization.create.masterData'"
+>
+  <div class="form-container">
+    <watt-form-field class="field-name">
+      <watt-label>{{ t("organizationNameLabel") }}</watt-label>
+      <input
+        wattInput
+        title="nameInput"
```
[the remark this answers is not in the document]
Comment (defectiveAi) [reply libs/dh/market-participant/feature-edit-organization/src/lib/dh-market-participant-edit-organization/master-data/dh-market-participant-organization-master-data.component.html:25]: Nope, we removed the title.

File: libs/dh/market-participant/feature-edit-organization/src/lib/dh-market-participant-edit-organization/master-data/dh-market-participant-organization-master-data.component.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+  constructor(private translocoService: TranslocoService) {}
+
+  changes: OrganizationChanges = { address: { country: 'DK' } };
+  countries: WattDropdownOption[] = [];
+
+  ngOnInit(): void {
+    const countryTranslations = this.translocoService.translateObject(
+      'marketParticipant.organization.create.masterData.countries'
+    );
```
[the remark this answers is not in the document]
Comment (defectiveAi) [reply libs/dh/market-participant/feature-edit-organization/src/lib/dh-market-participant-edit-organization/master-data/dh-market-participant-organization-master-data.component.ts:70]: Fixed, however, it appears the component does not refresh when the options are changed, unless the dropdown is opened and closed again.

File: libs/dh/market-participant/feature-edit-organization/src/lib/dh-market-participant-edit-organization/contact-data/dh-market-participant-organization-contact-data.component.html [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  *transloco="let t; read: 'marketParticipant.organization.create.contacts'"
+>
+  <mat-table [dataSource]="this.contactRows" aria-label="contacts">
+    <!-- Contact Type -->
+    <ng-container matColumnDef="{{ this.columnIds[0] }}">
+      <mat-header-cell *matHeaderCellDef>{{ t("category") }}</mat-header-cell>
+      <mat-cell *matCellDef="let element" class="field">
+        <watt-form-field>
+          <watt-dropdown
+            required
```
[the remark this answers is not in the document]
Comment (defectiveAi) [reply libs/dh/market-participant/feature-edit-organization/src/lib/dh-market-participant-edit-organization/contact-data/dh-market-participant-organization-contact-data.component.html:27]: No, the red border is enough. We think it looks weird inside the table.

File: libs/dh/market-participant/feature-edit-organization/src/lib/dh-market-participant-edit-organization/contact-data/dh-market-participant-organization-contact-data.component.html [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    </ng-container>
+
+    <!-- Name -->
+    <ng-container matColumnDef="{{ this.columnIds[1] }}">
+      <mat-header-cell *matHeaderCellDef>{{ t("name") }}</mat-header-cell>
+      <mat-cell *matCellDef="let element" class="field">
+        <watt-form-field
+          ><input
+            wattInput
+            required
```
[the remark this answers is not in the document]
Comment (defectiveAi) [reply libs/dh/market-participant/feature-edit-organization/src/lib/dh-market-participant-edit-organization/contact-data/dh-market-participant-organization-contact-data.component.html:44]: No, the red border is enough. We think it looks weird inside the table.

File: libs/dh/market-participant/feature-edit-organization/src/lib/dh-market-participant-edit-organization/contact-data/dh-market-participant-organization-contact-data.component.html [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    </ng-container>
+
+    <!-- Email -->
+    <ng-container matColumnDef="{{ this.columnIds[2] }}">
+      <mat-header-cell *matHeaderCellDef>{{ t("email") }}</mat-header-cell>
+      <mat-cell *matCellDef="let element" class="field">
+        <watt-form-field
+          ><input
+            wattInput
+            required
```
[the remark this answers is not in the document]
Comment (defectiveAi) [reply libs/dh/market-participant/feature-edit-organization/src/lib/dh-market-participant-edit-organization/contact-data/dh-market-participant-organization-contact-data.component.html:60]: No, the red border is enough. We think it looks weird inside the table.

File: libs/dh/market-participant/data-access-api/src/lib/dh-market-participant-data-access-api.store.ts [hunk 8]
```diff
@@ -15,55 +15,79 @@
...
 
 @Injectable()
 export class DhMarketParticipantOverviewDataAccessApiStore extends ComponentStore<MarketParticipantState> {
+  isLoading$ = this.select((state) => state.isLoading);
+  overviewList$;
+  validationError$ = this.select((state) => state.validation);
+
   constructor(private httpClient: MarketParticipantHttp) {
     super(initialState);
+    this.overviewList$ = this.setupRefreshListFlow();
```
[the remark this answers is not in the document]
Comment (defectiveAi) [reply libs/dh/market-participant/data-access-api/src/lib/dh-market-participant-data-access-api.store.ts]: Refactored, so the method no longer exist.

**Review by dzhavat — CHANGES_REQUESTED** [review]

**Review by dzhavat — APPROVED** [review]

**Pull request merged**